packages/provider/src/transports/window-transport/window-message-handler.ts

```diff
@@ -30,23 +33,41 @@ export class WindowMessageHandler extends BaseWalletTransport {
 
     // record open details (sessionId + default network) from the window url
     const location = new URL(window.location.href)
-    this._sessionId = sanitizeNumberString(location.searchParams.get('sid')!)
-    location.searchParams.delete('sid')
+
+    // TODO: add typings
+    const params = Object.fromEntries(new URLSearchParams(location.search))
+
+    this._sessionId = sanitizeNumberString(params['sid'])
 
     if (this._sessionId.length === 0) {
       logger.error('invalid sessionId')
       return
     }
 
-    const defaultNetwork = location.searchParams.get('net')!
-    location.searchParams.delete('net')
-
-    const jsonRpcRequest = location.searchParams.get('jsonRpcRequest')
-
-    if (options?.loadingPath && !!jsonRpcRequest) {
-      window.history.replaceState({}, document.title, options.loadingPath)
+    const defaultNetwork = params['net']
+
+    const toBool = (s: string) => s === 'true'
+
+    if (toBool(params['requestAuthorization'])) {
+      this.walletRequestHandler
```

Comment:
the WindowMessageHandler should not be calling `this.walletRequestHandler` ever

`this.walletRequestHandler` should be called from the base class of `BaseWalletTransport` so other implementations will follow the same code-path. Ie. proxy-transport and soon post-message-transport (for chrome extension)